File: dataset/output/CLAUDE_claude-3-5-sonnet-20240620/Task113_CLAUDE_claude-3-5-sonnet-20240620.java

import java.util.Scanner;

public class StringConcatenation {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        StringBuilder result = new StringBuilder();
        
        System.out.println("Enter strings to concatenate (type \'done\' to finish):");
        
        while (true) {
            String input = scanner.nextLine();
            if (input.equalsIgnoreCase("done")) {
                break;
            }
            result.append(input);
        }
        
        System.out.println("Concatenated string: " + result.toString());
        scanner.close();
    }
}

File: dataset/output-bk/CLAUDE_claude-3-5-sonnet-20240620/Task113_CLAUDE_claude-3-5-sonnet-20240620.c

#include <stdio.h>
#include <stdlib.h>
#include <string.h>

#define MAX_LENGTH 1000

int main() {
    char result[MAX_LENGTH] = "";
    char input[MAX_LENGTH];
    
    printf("Enter strings to concatenate (type \'done\' to finish):\
");
    
    while (1) {
        fgets(input, MAX_LENGTH, stdin);
        input[strcspn(input, "\
")] = 0;  // Remove newline
        
        if (strcmp(input, "done") == 0) {
            break;
        }
        
        strcat(result, input);
    }
    
    printf("Concatenated string: %s\
", result);
    
    return 0;
}
